internals/get-package-json/src/index.ts

```diff
@@ -0,0 +1,41 @@
+import fs from 'fs';
+import path from 'path';
+
+function getPackageJSONPath(dir: string) {
+  return path.join(dir, 'package.json');
+}
+
+export function getPackageJSON() {
+  const _prepareStackTrace = Error.prepareStackTrace;
+
+  Error.prepareStackTrace = (_, stack) => stack;
+
+  // The `stack` reference above is of type `Array<NodeJS.CallSite>`
+  const stack = (new Error().stack as unknown as Array<NodeJS.CallSite>).slice(
+    1
+  ); // remove self from stack
+
+  Error.prepareStackTrace = _prepareStackTrace;
+
+  // TypeScript rule noUncheckedIndexedAccess makes `stack[0]` potentially undefined;
+  // however, we can safely assert it is not undefined as the callstack will always
+  // have atleast 2 entries (we slice the first above), and then rely on the second here.
+  const callsite = stack[0]!;
```

Comment:
Why do you need to `stack.slice(1)` and then `stack[0]`.

Could you instead use `stack[1]` and then you don't need to slice?